filebeat/crawler/prospector.go

```diff
@@ -28,9 +31,24 @@ type Prospectorer interface {
 	Run()
 }
 
-func NewProspector(prospectorConfig cfg.ProspectorConfig, registrar *Registrar, spoolerChan chan *input.FileEvent) (*Prospector, error) {
+type prospectorConfig struct {
```

Comment:
I like to put these into separate config.go file.

Also add `func (p *prospectorConfig) Validate() error { return nil }`. This method is executed by ucfg to capture additional config failures and report these errors on Unpack. If error is reported, ucfg adds some additional context with path to hint which prospector config failed. In future ucfg might try to collect mutliple errors for reporting (think `-configtest`).